src/dags/entry_node.cpp

```diff
@@ -74,7 +76,9 @@ Status EntryNode<RequestType>::fetchResults(TensorWithSourceMap& outputs) {
     }
     InputSink<TensorWithSourceMap&> inputSink(outputs);
     bool isPipeline = true;
-    return deserializePredictRequest<ConcreteTensorProtoDeserializator>(*request, inputsInfo, inputSink, isPipeline);
+    std::unordered_map<int, std::shared_ptr<IOVTensorFactory>> factories;
+    factories.emplace(OVMS_BUFFERTYPE_CPU, std::make_shared<RegularOVTensorFactory>());  // TODO enable dag with remote tensors?
```

Comment:
It is performance critical path, is it good idea to re-create this map everytime? Why can't we have one global map for mapping between buffertype and factory, and just use buffer type to select factory from this single global map? Why cant we have one factory?

Reply:
Ok I see now, the factories contain context of the particular model, so it cannot be global and shared across for all servables

Reply:
It is ok for CPU buffers, but it would not be ok for GPU. Since we only support CPU for now for DAGs, we don't need to expose those factories outside of model instance. I will change that as suggested.